pkg/server/sotw/v3/xds.go

```diff
@@ -45,6 +45,10 @@ func (s *server) process(str stream.Stream, reqCh chan *discovery.DiscoveryReque
 	// s.ctx.Done()
 	sw.watches.recompute(s.ctx, reqCh)
 
+	if s.opts.Ordered {
```

Comment:
Do we care about the default case? 

```
		default:
			// Channel n -> these are the dynamic list of responders that correspond to the stream request typeURL
			if !ok {
				// Receiver channel was closed. TODO(jpeach): probably cancel the watch or something?
				return status.Errorf(codes.Unavailable, "resource watch %d -> failed", index)
			}
			res := value.Interface().(cache.Response)
			nonce, err := sw.send(res)
			if err != nil {
				return err
			}
			sw.watches.responders[res.GetRequest().GetTypeUrl()].nonce = nonce
		}```

Reply:
We do handle the default case IMHO. Default case is basically this
```
case res := <-sw.watches.responders[resource.AnyType].response:
			if err := process(res); err != nil {
				return status.Errorf(codes.Unavailable, "%v", err)
			}
```
this `!ok` is handled here
```
		case req, ok := <-reqCh:
			// Input stream ended or failed.
			if !ok {
				return nil
			}
```